docs/management-sdk-migration.md

```diff
@@ -0,0 +1,125 @@
+    .BuildServiceManager(); // note this
+```
+
+<!--Add sharding link-->
+### Create service hub context
+
+**Legacy APIs**
+```cs
+var serviceHubContext = await serviceManager.CreateServiceHubContextAsync("<Your Hub Name>", loggerFactory, cancellationToken);
+```
+
+**New APIs**
+```cs
+var serviceHubContext = await serviceManager.CreateServiceHubContextAsync("<Your Hub Name>", cancellationToken);
+```
+
+If you have custom logger factory, you should specify it when you create the service manager with new APIs. [See sample in the previous section](#build-service-manager)
+### Negotiation
+
+**Legacy APIs**
+```cs
+var clientEndpoint = serviceManager.GetClientEndpoint("<Your Hub Name>");
+var accessToken = serviceManager.GenerateClientAccessToken("<Your Hub Name>", "<Your User ID>");
+```
+
+**New APIs**
+```cs
+var negotiationResponse = await serviceHubContext.NegotiateAsync(new NegotiationOptions(){UserId = "<Your User Id>"});
+var clientEndpoint = negotiationResponse.Url;
+var accessToken = negotiationResponse.AccessToken;
+```
+The negotiation API is async now because there is an async operation behind if you use AAD connection string. In the old negotiation API, you just wait for the result synchronously.
+
+Except for the API change, there is also an implementation change. In [Persistent mode](management-sdk-guide.md#transport-type), you will call negotiation API after trying to establish WebSocket connections to Azure SignalR Service, so that you know whether the SignalR Service is healthy/online through the connection status. Therefore, the negotiation API only returns an healthy SignalR endpoint, if there is no healthy SignalR endpoint, an `AzureSignalRNotConnectedException` is thrown. In [Transient mode](management-sdk-guide.md#transport-type), we also have a plan to implement an endpoint health checker.
+
+### Send Messages and Manage Groups
+**Legacy APIs**
+```cs
+try
+{
+    var hubcontext = await serviceManager.CreateHubContextAsync("<Your Hub Name>");
+
+    // Broadcast
+    await hubContext.Clients.All.SendAsync(callbackName, obj1, obj2, ...);
+    
+    //...
+    
+    // add user to group
+    await hubContext.UserGroups.AddToGroupAsync(userId, groupName);
+}
+finally
+{
+    await hubContext.DisposeAsync();
+}
+```
+
+**New APIs**
+```cs
+try
+{
+    // Broadcast
+    await hubContext.Clients.All.SendAsync(callbackName, obj1, obj2, ...);
+
+    // add user to group
+    await hubContext.UserGroups.AddToGroupAsync(userId, groupName);
+}
+finally
+{
+    await hubContext.DisposeAsync();
+}
+```
+
+With legacy APIs, you need to create a hub context from `IServiceManager`, but with new APIs, you have built a hub context from `ServiceHubContextBuilder` at the beginning. The usages of hub context are nearly the same, except that in the legacy APIs you get an interface `IServiceHubContext` while in the new APIs you get an abstract class `ServiceHubContext`.
```

Comment:
> With legacy APIs, you need to create a hub context from `IServiceManager`, but with new APIs, you have built a hub context from `ServiceHubContextBuilder` at the beginning.

why the legacy one can't create the hub context from the beginning?

Reply:
Outdated.